### ISSUE TITLE
Remove named datasets, in favour of referring by index instead

### ISSUE THREAD

**Issue by yousefmoazzam** [DiamondLightSource/httomo#171]
Due to the nature of the GPU loop overwriting input arrays with the result of the output of a method/section, the naming of datasets is no longer really necessary.

For example, if the dataset loaded by the loader was called `tomo`, and the `tomo` dataset was overwritten with the result of some methods, no situation comes to mind where a user would _not_ want to pass the `tomo` dataset to later methods.

For cases where a method returns multiple outputs, referring to output datasets by index rather than by a name appears to be sufficient (ie, specify which method number, and which output dataset).

This work is planned to begin once the `gpuloop` branch is merged into `main`.

**Comment by yousefmoazzam** [comment]
This is obsolete, we now have "side outputs" in the framework to deal with when a method needs to generate "extra/supplementary outputs".
